src/System.Runtime.Environment/tests/CheckPlatformTests.cs

```diff
@@ -0,0 +1,41 @@
+﻿using System.Runtime.InteropServices;
+using Xunit;
+
+namespace System.Runtime.Environment.Tests
+{
+    public class CheckPlatformTests
+    {
+        [Fact, PlatformSpecific(PlatformID.Windows)]
+        public void CheckWindows()
+        {
+            Assert.True(RuntimeInfo.IsOSPlatform(OSName.Windows));
+            Assert.True(RuntimeInfo.IsOSPlatform(new OSName("WINDOWS")));
```

Comment:
If it was me, I would not support the behavior of case insensitively or treating Unix as Linux, etc.